wgsl/extract-grammar.py

```diff
@@ -255,118 +338,287 @@ def parse(lines, i):
+                if production["value"] in result[scanner_token.name()]:
+                    subsource = f"<a for=syntax_sym lt={result[scanner_token.name()][production['value']]}>`{production['value']}`</a>"
+                else:
+                    subsource = f"`{production['value']}`"
+            elif production["type"] == "pattern":
+                subsource = f"`{production['value']}`"
+            return subsource
+        elif production["type"] == "sequence":
+            if len(rule_name) == 0:
+                return (
+                    "( "
+                    + " ".join(
+                        [
+                            reproduce_rule_source(member)
+                            for member in production["value"]
+                        ]
+                    )
+                    + " )"
+                )
+            else:
+                return " ".join(
+                    [reproduce_rule_source(member) for member in production["value"]]
+                )
+        elif production["type"] == "choice":
+            return (
+                "( "
+                + " | ".join(
+                    [reproduce_rule_source(member) for member in production["value"]]
+                )
+                + " )"
+            )
+        elif production["type"] == "match_0_1":
+            if len(production["value"]) == 1:
+                return reproduce_rule_source(production["value"][0]) + " ?"
+            else:
+                return (
+                    "( "
+                    + " ".join(
+                        [
+                            reproduce_rule_source(member)
+                            for member in production["value"]
+                        ]
+                    )
+                    + " ) ?"
+                )
+        elif production["type"] == "match_1_n":
+            if len(production["value"]) == 1:
+                return reproduce_rule_source(production["value"][0]) + " +"
+            else:
+                return (
+                    "( "
+                    + " ".join(
+                        [
+                            reproduce_rule_source(member)
+                            for member in production["value"]
+                        ]
+                    )
+                    + " ) +"
+                )
+        elif production["type"] == "match_0_n":
+            if len(production["value"]) == 1:
+                return reproduce_rule_source(production["value"][0]) + " *"
+            else:
+                return (
+                    "( "
+                    + " ".join(
+                        [
+                            reproduce_rule_source(member)
+                            for member in production["value"]
+                        ]
+                    )
+                    + " ) *"
+                )
+        else:
+            print(production["type"])
+
+    if value["type"] in ["symbol", "token", "pattern"]:
```

Comment:
Would save typing and reading if you used an intermediate variable for value["type"]

Similar issue in a few of the other functions.

Reply:
We should probably align to a linter for such choices. I've tried variable names like type_v but didn't feel as readable to me. Are there any suggested variable names if it provides a clear readability gain?